File: 백준/Bronze/5585. 거스름돈/거스름돈.py
cost = int(input())
change = 1000 - cost

def min_coins(change):
    coins = [500, 100, 50, 10, 5, 1]
    count = 0
    
    rest = change
    while rest > 0:
        for coin in coins:
            if coin <= rest:
                rest -= coin
                count += 1
                break
    return count
    
print(min_coins(change))

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int cost = sc.nextInt();
        sc.close();
        
        int change = 1000 - cost;
        int[] coins = {500, 100, 50, 10, 5, 1};
        int count = 0;
        
        for (int coin : coins) {
            count += change / coin;
            change %= coin;
        }
        System.out.print(count);
    }
}